jobs/build-microk8s-charm.yaml

```diff
@@ -0,0 +1,94 @@
+# Builds microk8s charm and uploads to CharmHub
+
+- job:
+    name: 'build-microk8s-charm'
+    description: |
+      Build MicroK8s charm and upload revision to CharmHub.
+    project-type: freestyle
+    node: runner-amd64
+    wrappers:
+      - ansicolor
+      - timestamps
+      - workspace-cleanup
+      - credentials-binding:
+          - file:
+              credential-id: JUJUCONTROLLERS_SERVERSTACK
```

Comment:
@johnsca @kwmonroe this job uses our internal serverstack. Do you see any issue with that?

Reply:
Only concern would be left-over resources, but it looks like the `cleanup` code in this PR should tear everything down nicely.